pwnlib/commandline/libcdb.py

```diff
@@ -136,17 +150,45 @@
 
 common_symbols = ['dup2', 'printf', 'puts', 'read', 'system', 'write']
 
-def find_libc(params):
-    import requests
-    url    = "https://libc.rip/api/find"
-    result = requests.post(url, json=params, timeout=20)
-    log.debug('Request: %s', params)
-    log.debug('Result: %s', result.json())
-    if result.status_code != 200 or len(result.json()) == 0:
-        log.failure("Could not find libc for %s on libc.rip", params)
-        return []
 
-    return result.json()
+def find_libc(params, offline=False):
```

Comment:
Directly using the internal `find_local_libc_database`, `get_libc_info`, and `get_libc_symbols` looks clunky :(
Can we add this offline lookup logic to `search_by_symbol_offsets` and add a `raw` parameter which returns the raw json result for further processing?

Then we don't need the `--offline` argument and always check the offline database if it's configured!

Reply:
>Directly using the internal `find_local_libc_database`, `get_libc_info`, and `get_libc_symbols` looks clunky :(

I haven't yet come up with a more elegant solution for merging the logic between online and offline searches, so I still believe that having a separate `find_local_libc_database` function to handle offline searches is necessary.

Regarding `get_libc_info`, It's a temporary solution designed to match the format of [libc.rip](libc.rip)'s output.

For `get_libc_symbols`, I will attempt to optimize this function.

>Can we add this offline lookup logic to `search_by_symbol_offsets` and add a `raw` parameter which returns the raw json result for further processing?

Adding a `raw` parameter is a feasible option.